### PR TITLE
data file should use path to judge if equal

### PR DESCRIPTION
**Pull request by adrian-wang**
## What changes were proposed in this pull request?

Otherwise the `DataFileMetaCache` will be loaded over and over again.

## How was this patch tested?

manual test.

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/io/DataFile.scala [hunk 1]
```diff
@@ -45,6 +45,8 @@ abstract class DataFile {
     : OapIterator[InternalRow]
 
   def totalRows(): Long
+  override def hashCode(): Int = path.hashCode
+  override def equals(other: Any): Boolean = path.equals(other.asInstanceOf[DataFile].path)
```
[the remark this answers is not in the document]
Comment (gczsjdy) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/io/DataFile.scala]: +1

File: src/test/scala/org/apache/spark/sql/execution/datasources/oap/io/DataFileSuite.scala [hunk 2]
```diff
@@ -70,4 +70,31 @@ class DataFileSuite extends QueryTest with SharedOapContext {
...
+    val config = new Configuration()
+    withTempPath { dir =>
+      val df = spark.createDataFrame(data)
+      df.repartition(1).write.parquet(dir.getAbsolutePath)
+      val file = SpecificParquetRecordReaderBase.listDirectory(dir).get(0)
+      val datafile1 =
+        DataFile(file, schema, OapFileFormat.PARQUET_DATA_FILE_CLASSNAME, config)
+      val datafile2 =
+        DataFile(file, schema, OapFileFormat.PARQUET_DATA_FILE_CLASSNAME, config)
+      assert(datafile1.equals(datafile2))
```
Comment (gczsjdy) [inline src/test/scala/org/apache/spark/sql/execution/datasources/oap/io/DataFileSuite.scala]: Also assert the hashCode equals?

Comment (adrian-wang) [reply src/test/scala/org/apache/spark/sql/execution/datasources/oap/io/DataFileSuite.scala]: done.

**Comment by adrian-wang** [comment]
Thanks @linhong-intel & @gczsjdy

**Pull request merged**